File: back/src/test/java/csh/back/domain/trip/group/support/WithMockLoginUser.java
package csh.back.domain.trip.group.support;

import org.springframework.security.test.context.support.WithSecurityContext;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = WithMockLoginUserFactory.class)
public @interface WithMockLoginUser {
	long id() default 1L;
	String email() default "admin@admin.com";
}

File: back/src/test/kotlin/csh/back/domain/trip/group/support/WithMockLoginUser.kt
package csh.back.domain.trip.group.support

import org.springframework.security.test.context.support.WithSecurityContext
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

@Retention(RetentionPolicy.RUNTIME)
@WithSecurityContext(factory = WithMockLoginUserFactory::class)
annotation class WithMockLoginUser(
    val id: Long = 1L,
    val email: String = "admin@admin.com"
)
